pkg/storage/tpr/storage_interface.go

```diff
@@ -106,8 +106,18 @@ func (t *store) Create(
 		t.singularKind.URLName(),
 	).Body(data)
 
+	res := req.Do()
+	if res.Error() != nil {
+		glog.Errorf("executing DELETE for %s/%s (%s)", ns, name, res.Error())
+	}
+	var statusCode int
+	res.StatusCode(&statusCode)
+	if statusCode == http.StatusConflict {
+		return storage.NewKeyExistsError(key, 0)
+	}
```

Comment:
That's not clear to me, tbh. Should we have a massive switch for handling all possible response codes? If this can be improved, I think it should be dealt with as a separate issue.